src/zennit/types.py

```diff
@@ -124,3 +126,10 @@ class Activation(metaclass=SubclassMeta):
         torch.nn.modules.activation.Tanhshrink,
         torch.nn.modules.activation.Threshold,
     )
+
+
+class Distance(metaclass=SubclassMeta):
+    '''Abstract base class that describes distance modules.'''
```

Comment:
I do not think this class is necessary, since this is not used for general distance modules, but rather only for the `PairwiseCentroidDistance` module, as the `centroids` are accessed after an instance-check in the canonizer.
Unless there is another use, this can be removed, and the canonizer can directly use `PairwiseCentroidDistance`.

Or will there be more centroid-based distance layer types?

Reply:
Agreed, can be removed. (will do)